### ISSUE TITLE
Add centerOffset property to MGLAnnotationView

### ISSUE THREAD

**Issue by boundsj** [mapbox/mapbox-gl-native#5059]
MapKit has a [centerOffset property available on MKAnnotationView](https://developer.apple.com/library/ios/documentation/MapKit/Reference/MKAnnotationView_Class/#//apple_ref/occ/instp/MKAnnotationView/centerOffset) that allows one "to reposition the annotation view as needed" from the default center point of the annotation model object itself. We should add this feature to MGLAnnotationView.

fyi @1ec5 @friedbunny @tobrun

**Comment by 1ec5** [comment]
Getting this functionality into MGLAnnotationImage is being tracked in #3220 / #2151, but that’s blocked by the need for a dynamic styling API. Getting it into MGLAnnotationView neatly avoids that mess.

**Comment by domasn** [comment]
@1ec5 It seems that you forgot to take `MGLAnnotationView.centerOffset` into account when [getting an annotation at point](https://github.com/mapbox/mapbox-gl-native/blob/45d65c4b491615e0ff0c1ec0a89324159bc1dc97/platform/ios/src/MGLMapView.mm#L3249).

This results in `didSelectAnnotation` delegate method being called unreasonably..

**Comment by 1ec5** [comment]
Good catch, @domasn! Please open a new issue for this case.

**Comment by friedbunny** [comment]
@domasn @1ec5 this issue was opened just now as https://github.com/mapbox/mapbox-gl-native/issues/5820.
